sbx/recurrent_ppo/policies.py

```diff
@@ -0,0 +1,415 @@
+import functools
+
+from dataclasses import field
+from typing import Any, Callable, Dict, List, Optional, Sequence, Union, Tuple
+
+import flax.linen as nn
+import gymnasium as gym
+import jax
+import jax.numpy as jnp
+import numpy as np
+import optax
+import tensorflow_probability.substrates.jax as tfp
+from flax.linen.initializers import constant
+from flax.training.train_state import TrainState
+from gymnasium import spaces
+from stable_baselines3.common.type_aliases import Schedule
+
+from sbx.common.policies import BaseJaxPolicy
+from sbx.common.recurrent import LSTMStates
+
+tfd = tfp.distributions
+
+
+# Added a ScanLSTM Module that automatically handles the reset of LSTM states
+# inspired from the ScanRNN in purejaxrl : https://github.com/luchris429/purejaxrl/blob/main/purejaxrl/ppo_rnn.py
+class ScanLSTM(nn.Module):
+    @functools.partial(
+        nn.scan,
+        variable_broadcast='params',
+        in_axes=0,
+        out_axes=0,
+        split_rngs={'params': False}
+    )
+    @nn.compact
+    def __call__(self, lstm_states, inputs_and_resets):
+        # pass the pi and vf lstm states, as well as the obs and the resets
+        input, resets = inputs_and_resets
```

Comment:
with `input`, you are shadowing a python built-in ... maybe `obs, resets` is better?